### PR TITLE
Add test-provision-gce

### PR DESCRIPTION
**Pull request by dkropachev**
https://trello.com/c/fqHEngaB/1375-add-gce-backend-check-to-ci

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [ ] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [ ] ~~All new and existing unit tests passed (CI)~~
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

**Comment by fruch** [comment]
Maybe using `load`, and separate helper functions into it's own groovy file
https://jenkins.io/doc/pipeline/steps/workflow-cps/#load-evaluate-a-groovy-source-file-into-the-pipeline-script

**Comment by dkropachev** [comment]
Currently experiance problem with gce provision :

18:35:25  Traceback (most recent call last):
18:35:25    File "/sct/sdcm/tester.py", line 250, in setUp
18:35:25      self.init_resources()
18:35:25    File "/sct/sdcm/tester.py", line 662, in init_resources
18:35:25      monitor_info=monitor_info)
18:35:25    File "/sct/sdcm/tester.py", line 410, in get_cluster_gce
18:35:25      **common_params)
18:35:25    File "/sct/sdcm/cluster_gce.py", line 370, in __init__
18:35:25      node_type='scylla-db'
18:35:25    File "/sct/sdcm/cluster.py", line 2663, in __init__
18:35:25      super(BaseScyllaCluster, self).__init__(*args, **kwargs)
18:35:25    File "/sct/sdcm/cluster_gce.py", line 175, in __init__
18:35:25      node_type=node_type)
18:35:25    File "/sct/sdcm/cluster.py", line 2462, in __init__
18:35:25      self.add_nodes(num, dc_idx=dc_idx)
18:35:25    File "/sct/sdcm/cluster_gce.py", line 328, in add_nodes
18:35:25      instances = self._create_instances(count, dc_idx)
18:35:25    File "/sct/sdcm/cluster_gce.py", line 276, in _create_instances
18:35:25      instances.append(self._create_instance(node_index, dc_idx))
18:35:25    File "/sct/sdcm/cluster_gce.py", line 265, in _create_instance
18:35:25      'NodeType': self.node_type})
18:35:25    File "/usr/local/lib/python3.6/site-packages/libcloud/compute/drivers/gce.py", line 4170, in create_node
18:35:25      self.connection.async_request(request, method='POST', data=node_data)
18:35:25    File "/usr/local/lib/python3.6/site-packages/libcloud/common/base.py", line 773, in async_request
18:35:25      response = request(**kwargs)
18:35:25    File "/usr/local/lib/python3.6/site-packages/libcloud/compute/drivers/gce.py", line 121, in request
18:35:25      response = super(GCEConnection, self).request(*args, **kwargs)
18:35:25    File "/usr/local/lib/python3.6/site-packages/libcloud/common/google.py", line 813, in request
18:35:25      *args, **kwargs)
18:35:25    File "/usr/local/lib/python3.6/site-packages/libcloud/common/base.py", line 638, in request
18:35:25      response = responseCls(**kwargs)
18:35:25    File "/usr/local/lib/python3.6/site-packages/libcloud/common/base.py", line 154, in __init__
18:35:25      self.object = self.parse_body()
18:35:25    File "/usr/local/lib/python3.6/site-packages/libcloud/common/google.py", line 285, in parse_body
18:35:25      raise InvalidRequestError(message, self.status, code)
18:35:25  libcloud.common.google.InvalidRequestError: {'domain': 'global', 'reason': 'invalid', 'message': "Invalid value for field 'resource.name': 'PR-provision-test-jenkins-db-node-4f22a047-0-1'. Must be a match of regex '(?:[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?)'"}

**Comment by dkropachev** [comment]
> Currently experiance problem with gce provision :
> 
> 18:35:25 Traceback (most recent call last):
> 18:35:25 File "/sct/sdcm/tester.py", line 250, in setUp
> 18:35:25 self.init_resources()
> 18:35:25 File "/sct/sdcm/tester.py", line 662, in init_resources
> 18:35:25 monitor_info=monitor_info)
> 18:35:25 File "/sct/sdcm/tester.py", line 410, in get_cluster_gce
> 18:35:25 **common_params)
> 18:35:25 File "/sct/sdcm/cluster_gce.py", line 370, in **init**
> 18:35:25 node_type='scylla-db'
> 18:35:25 File "/sct/sdcm/cluster.py", line 2663, in **init**
> 18:35:25 super(BaseScyllaCluster, self).**init**(*args, **kwargs)
> 18:35:25 File "/sct/sdcm/cluster_gce.py", line 175, in **init**
> 18:35:25 node_type=node_type)
> 18:35:25 File "/sct/sdcm/cluster.py", line 2462, in **init**
> 18:35:25 self.add_nodes(num, dc_idx=dc_idx)
> 18:35:25 File "/sct/sdcm/cluster_gce.py", line 328, in add_nodes
> 18:35:25 instances = self._create_instances(count, dc_idx)
> 18:35:25 File "/sct/sdcm/cluster_gce.py", line 276, in _create_instances
> 18:35:25 instances.append(self._create_instance(node_index, dc_idx))
> 18:35:25 File "/sct/sdcm/cluster_gce.py", line 265, in _create_instance
> 18:35:25 'NodeType': self.node_type})
> 18:35:25 File "/usr/local/lib/python3.6/site-packages/libcloud/compute/drivers/gce.py", line 4170, in create_node
> 18:35:25 self.connection.async_request(request, method='POST', data=node_data)
> 18:35:25 File "/usr/local/lib/python3.6/site-packages/libcloud/common/base.py", line 773, in async_request
> 18:35:25 response = request(**kwargs)
> 18:35:25 File "/usr/local/lib/python3.6/site-packages/libcloud/compute/drivers/gce.py", line 121, in request
> 18:35:25 response = super(GCEConnection, self).request(*args, **kwargs)
> 18:35:25 File "/usr/local/lib/python3.6/site-packages/libcloud/common/google.py", line 813, in request
> 18:35:25 *args, **kwargs)
> 18:35:25 File "/usr/local/lib/python3.6/site-packages/libcloud/common/base.py", line 638, in request
> 18:35:25 response = responseCls(**kwargs)
> 18:35:25 File "/usr/local/lib/python3.6/site-packages/libcloud/common/base.py", line 154, in **init**
> 18:35:25 self.object = self.parse_body()
> 18:35:25 File "/usr/local/lib/python3.6/site-packages/libcloud/common/google.py", line 285, in parse_body
> 18:35:25 raise InvalidRequestError(message, self.status, code)
> 18:35:25 libcloud.common.google.InvalidRequestError: {'domain': 'global', 'reason': 'invalid', 'message': "Invalid value for field 'resource.name': 'PR-provision-test-jenkins-db-node-4f22a047-0-1'. Must be a match of regex '(?:a-z?)'"}

Fixed

File: Jenkinsfile [hunk 1]
```diff
@@ -1,136 +1,148 @@
...
+      failFast true
+      parallel {
+        stage('provision test on gce') {
+          when {
+            expression {
+              return lookForLabels("test-provision,test-provision-gce")
+            }
+          }
+          steps {
             node('aws-sct-builders-eu-west-1') {
```
Comment (fruch) [inline Jenkinsfile]: Use gce builders. So it won't be cross cloud setup. We don't want to pay that extra on the traffic

File: Jenkinsfile [hunk 2]
```diff
@@ -1,136 +1,148 @@
...
     stage("provision test") {
-        when { changeRequest() }
-        steps {
+      when {
+        expression {
+          return lookForLabels("test-provision,test-provision-aws,test-provision-gce") && currentBuild.result == null
+        }
+      }
+      failFast true
+      parallel {
```
Comment (fruch) [inline Jenkinsfile]: :+1: for using parallel here, I would also add the docker backend when it's ready

**Comment by bentsi** [comment]
@dkropachev please address comments and check why provision test failed

File: Jenkinsfile (hunk 1 above)
Comment (dkropachev) [reply Jenkinsfile]: Fixed

File: Jenkinsfile (hunk 2 above)
Comment (dkropachev) [reply Jenkinsfile]: We will add, but we will need to keep docker host somewhere in the cloud for these purposes.

File: Jenkinsfile [hunk 3]
```diff
@@ -1,136 +1,217 @@
...
+	#!/bin/bash
+	set -xe
+	env
+	export SCT_CLUSTER_BACKEND="${backend}"
+	export SCT_COLLECT_LOGS=false
+	export SCT_POST_BEHAVIOR_DB_NODES="destroy"
+	export SCT_POST_BEHAVIOR_LOADER_NODES="destroy"
+	export SCT_POST_BEHAVIOR_MONITOR_NODES="destroy"
+	export SCT_INSTANCE_PROVISION="on_demand"
+    export SCT_CONFIG_FILES="test-cases/PR-provision-test.yaml1"
```
Comment (bentsi) [inline Jenkinsfile]: typo in the filename: `test-cases/PR-provision-test.yaml1` -> `test-cases/PR-provision-test.yaml`

Comment (dkropachev) [reply Jenkinsfile]: Fixed

**Pull request merged**